docs/components/tree-list.md

```diff
@@ -0,0 +1,330 @@
+  <KTreeList v-model="myList" />
+  <br>
+  <KButton @click="reset">Reset</KButton>
+</div>
+
+```html
+<template>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <KButton @click="reset">Reset</KButton>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data() {
+    return {
+      myList: [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  },
+  methods: {
+    reset() {
+      this.myList = [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  }
+})
+</script>
+```
+
+## KTreeItem
+
+### Properties
+
+- `name` (required) - text displayed as the label for the item
+- `id` (required) - a unique value used to identify the item
+- `selected` - boolean to indicate whether the current item is selected or not
+- `icon` - `KIcon` name to be displayed to the left of the item `name` (defaults to `treeDoc`, specify `none` to not display any icon)
+- `children` - an array of `KTreeItem`'s that will be styled as children of the current item
+
+### Item Slots
+
+- `item-icon` - slot for content displayed to the left of the item name
+- `item-label` - slot for the main content of an item (default's to display the `name` of the item)
+
+See the [Slots section](#Slots) for an example.
+
+## Slots
+
+`KTreeList` allows you to customize individual tree items via the item slots. These slots return the current `item` data as a slot param.
+
+- `item-icon` - slot for content displayed to the left of the item name
+- `item-label` - slot for the main content of an item (default's to display the `name` of the item)
+
+<KTreeList :items="slotItems">
+  <template #item-icon="{ item }">
+    {{ item.id === 'cats' ? '😸' : item.id === 'bunnies' ? '🐰' : '🐶' }}
+  </template>
+  <template #item-label="{ item }">
+    <span class="color-purple-400">
+    {{ item.name }}
+    </span>
+  </template>
+</KTreeList>
+
+```html
+<KTreeList :items="items">
+  <template #item-icon="{ item }">
+    {{ item.id === 'cats' ? '😸' : item.id === 'bunnies' ? '🐰' : '🐶' }}
+  </template>
+  <template #item-label="{ item }">
+    <span class="color-purple-400">
+    {{ item.name }}
+    </span>
+  </template>
+</KTreeList>
+```
+
+## Events
+
+- `@change` - Emitted when there is a change to the root level items
+  - returns `items` - an array of `KTreeItem`s
+- `@child-change` - Emitted when an item is added or removed at the non-root level
+  - returns `parent` - id of the parent item; `children` - an array of `KTreeItem`s
```

Comment:
question: what if the parent doesn't have an `id`?

Reply:
All items are required to have a `name` and `id`, so it will.